src/dnx.coreclr.unix/dnx.coreclr.cpp

```diff
@@ -0,0 +1,229 @@
+#include "stdafx.h"
+#include "tpa.h"
+#include <assert.h>
+
+// Windows types used by the ExecuteAssembly function
+typedef int32_t HRESULT;
+typedef const char* LPCSTR;
+typedef uint32_t DWORD;
+
+typedef struct CALL_APPLICATION_MAIN_DATA
+{
+    char* applicationBase;
+    char* runtimeDirectory;
+    int argc;
+    char** argv;
+    int exitcode;
+} *PCALL_APPLICATION_MAIN_DATA;
+
+// Prototype of the ExecuteAssembly function from the libcoreclr.do
+typedef HRESULT (*ExecuteAssemblyFunction)(
+                    LPCSTR exePath,
+                    LPCSTR coreClrPath,
+                    LPCSTR appDomainFriendlyName,
+                    int propertyCount,
+                    LPCSTR* propertyKeys,
+                    LPCSTR* propertyValues,
+                    int argc,
+                    LPCSTR* argv,
+                    LPCSTR managedAssemblyPath,
+                    LPCSTR entryPointAssemblyName,
+                    LPCSTR entryPointTypeName,
+                    LPCSTR entryPointMethodsName,
+                    DWORD* exitCode);
+
+const HRESULT S_OK = 0;
+const HRESULT E_FAIL = -1;
+
+bool GetTrustedPlatformAssembliesList(const std::string& tpaDirectory, bool isNative, std::string& trustedPlatformAssemblies)
+{
+    size_t cTpaAssemblyNames = 0;
+    LPCTSTR* ppszTpaAssemblyNames = nullptr;
+
+    CreateTpaBase(&ppszTpaAssemblyNames, &cTpaAssemblyNames, isNative);
+
+    assert(ppszTpaAssemblyNames != nullptr || cTpaAssemblyNames == 0);
+
+    //TODO: The Windows version of this actaully ensures the files are present.  We just fail for native and assume MSIL is present
+    if (isNative)
```

Comment:
I'm going to push back against this.  We need to remove the if (isNative) check in favor of doing the same thing for both native and the msil cases, which is looking on the file system to ensure all of the files from the TPA list are present.  For now, we just use the isNative check as a heuristic instead of actually probing the disk.  So long term the "should we return true or false from this method" decision is going to be made starting around L48.  If we move the code now, we'll just have to delete it later and it gets things further away from what we want, IMHO.